### ISSUE TITLE
'charts' visualization - possible client problems

### ISSUE THREAD

**Issue by martenson** [galaxyproject/galaxy#5736]
I have now seen multiple reports and can confirm myself.

*This is the error on `release_18.01`.*
![screenshot 2018-03-20 11 13 43](https://user-images.githubusercontent.com/1814954/37663553-cfde0232-2c2f-11e8-93c9-f5965dca7de1.png)

*These are the errors on `dev`*
edit: errors on dev moved to https://github.com/galaxyproject/galaxy/issues/5738

I suspect some of this can be remedied by running a proper makefile target, but given Galaxy builds client on the fly it should also build charts on the fly. Also `make charts` fails for me with :

```
webpack -p --config ../config/plugins/visualizations/charts/webpack.config.js
module.js:529
    throw err;
    ^

Error: Cannot find module '/Users/marten/devel/git/galaxy/config/plugins/visualizations/charts/webpack.config.js'
    at Function.Module._resolveFilename (module.js:527:15)
    at Function.Module._load (module.js:476:23)
    at Module.require (module.js:568:17)
    at require (internal/module.js:11:18)
    at requireConfig (/Users/marten/devel/git/galaxy/client/node_modules/webpack/bin/convert-argv.js:97:18)
    at /Users/marten/devel/git/galaxy/client/node_modules/webpack/bin/convert-argv.js:104:17
    at Array.forEach (<anonymous>)
    at module.exports (/Users/marten/devel/git/galaxy/client/node_modules/webpack/bin/convert-argv.js:102:15)
    at yargs.parse (/Users/marten/devel/git/galaxy/client/node_modules/webpack/bin/webpack.js:171:41)
    at Object.Yargs.self.parse (/Users/marten/devel/git/galaxy/client/node_modules/webpack/node_modules/yargs/yargs.js:533:18)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
make: *** [charts] Error 1
```

edit: the MSA-related issue extracted to https://github.com/galaxyproject/galaxy/issues/5739

**Comment by dannon** [comment]
@martenson @guerler Taking a look now (at 18.01 -- charts code has been significantly overhauled in dev)

**Comment by dannon** [comment]
@martenson Can you elaborate on the non-msa issue you're seeing on main (the 18.01 error indicated above)?  When are you seeing this?

**Comment by martenson** [comment]
@dannon that is not on Main, that is release_18.01 locally. I see it after clicking on visualize/charts button on a `txt` dataset

**Comment by dannon** [comment]
@martenson Ok, sorry, the issue report above was confusing to me.  Charts should be built correctly for 18.01 already -- the only file missing would be the .map, which is useless in this case.

@guerler and I are talking about the MSA viewer; it looks like we can change the specificity of the expected filetype to something like 'fasta' + 'phylip' and that'll improve that interaction.

**Comment by guerler** [comment]
@martenson the MSA error looks correct. Currently MSA viewer is shown when a txt file is selected. The correct format would be ClustalW. However the other error ("Load function not found!") is a problem. It looks like a build conflict. In current dev, the charts ui is part of the visualization framework and is build through "make client". In 18.01 it is still separated and build separately.

Edit: @martenson can you try to clear your cache and rebuild?

**Comment by guerler** [comment]
Duplicate #5738?

**Comment by martenson** [comment]
the issue on `release_18.01` branch still stays

**Comment by martenson** [comment]
After re-testing on 18.01 with Paste as the webserver I agree the first part is a duplicate (giving different errors but for the same reasons). The only console errors that I see on Paste is 

```
require.js?v=1521568974:1 Uncaught Error: Script error for "underscore"
http://requirejs.org/docs/errors.html#scripterror
    at makeError (require.js?v=1521568974:1)
    at HTMLScriptElement.onScriptError (require.js?v=1521568974:1)
```

which does not seem to prevent charts functionality.

The second part has been resolved with #5737.

Closing.
